Automattic/woocommerce-payments
# WooCommerce Deposits Multi Currency compatibility

Fixes #1902

#### Changes proposed in this Pull Request


This PR adds compatibility between WooCommerce Payments Multi-currency and WooCommerce Deposits Plugin. With this PR, the customers can buy products with splittable payment options with any currency the merchant enabled for their site. Before this PR:
- The deposit amount wasn't calculated in the correct currency
- The future orders were created with the store currency, not with the original order currency.

#### Testing instructions

##### Product with a single deposit option

- **Preparation Steps**

  * Install the WooCommerce Deposits plugin and activate it.
  * Go to one of the products' admin page and see the "Deposits" tab on the general product settings section.
  * Enable deposits for that product 
  * Select "Deposit Type" as "Percentage" or "Fixed Amount"
  * Enter a fixed amount or percentage relative to your selection on the previous step
  * Update the product by clicking the "Update" button
  
- **Testing**
  * Go to the shop page and sell the item
  * Check the prices on the cart dropdown and the cart page
  * Go to checkout and check the prices
  * Make the purchase and check the prices on the confirmation page
  * Go to WooCommerce > Orders page, select the order you just created
  * Check the prices there, and also you should see a "Invoice remaining balance" button
  * Click it. Check the prices and the currency on the new order created for the remaining amount.
  

##### Product with a payment plan type deposit option

- **Preparation Steps**

  * Create a payment plan on "Admin > Products > Payment Plans" page, for testing the prices easily, you can select percentages like 10%, 50% etc. 
  * After you've created the payment plan, go to one of the products' admin page and see the "Deposits" tab on the general product settings section.
  * Enable deposits for that product 
  * Select "Deposit Type" as "Payment Plan"
  * Select the payment plan you just created.
  * Update the product by clicking the "Update" button
  
- **Testing**
  * Go to the shop page and sell the item
  * Check the prices on the cart dropdown and the cart page
  * Go to checkout and check the prices
  * Make the purchase and check the prices on the confirmation page
  * Go to WooCommerce > Orders page, select the order you just created
  * Check the prices there, and also you should see a "View Scheduled Payments" button
  * Click it. Check the scheduled orders for the remaining payment steps.

**Extra testing:** Create a discount coupon and redo the tests with a coupon entered in the cart page

-------------------


**Post merge**
